sdk/identity/identity/test/internal/node/msalClient.spec.ts

```diff
@@ -11,8 +11,39 @@ import { IdentityClient } from "../../../src/client/identityClient";
 import { assert } from "@azure/test-utils";
 import { credentialLogger } from "../../../src/util/logging";
 import sinon from "sinon";
+import { MsalTestCleanup, msalNodeTestSetup } from "../../node/msalNodeTestSetup";
+import { Recorder } from "@azure-tools/test-recorder";
 
 describe("MsalClient", function () {
+  describe("recorded tests", function () {
+    let cleanup: MsalTestCleanup;
+    let recorder: Recorder;
+
+    afterEach(async function () {
+      await cleanup();
+    });
+
+    beforeEach(async function () {
+      ({ cleanup, recorder } = await msalNodeTestSetup(this.currentTest, "azure_client_id"));
+    });
+
+    it("supports getTokenByClientSecret", async function () {
```

Comment:
Not all tests will play well with recordings, but at least we now have a place to record and playback general-purpose tests that ensure the thing works. We can expand this later as needed. The nice thing is that the mock-based tests do not need to use this `msalNodeTestSetup` helper, simplifying them.

All we need to get recorded tests working is to feed tokenCredentialOptions which already exists with the additional recorder policies, and they get plumbed through to MSAL